tests/digraph/test_tensor_product.py

```diff
@@ -0,0 +1,60 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+import retworkx
+
+
+class TestTensorProduct(unittest.TestCase):
+    def test_directed_null_tensor_null(self):
+        graph_1 = retworkx.PyDiGraph()
+        graph_2 = retworkx.PyDiGraph()
+
+        graph_product, _ = retworkx.digraph_tensor_product(graph_1, graph_2)
```

Comment:
It might be good to check the output `ProductNodeMap` is correct too.

Reply:
Added in [`c2795ad`](https://github.com/Qiskit/retworkx/pull/472/commits/c2795ade2c1278430503f2757a58891ec9c326b0)